addons/stock_account/models/product.py

```diff
@@ -8,17 +8,55 @@
 from collections import defaultdict
 from datetime import datetime
 
+from odoo.addons.stock_account import const
+
 
 class ProductTemplate(models.Model):
     _inherit = ['product.template']
 
-    cost_method = fields.Selection(related="categ_id.property_cost_method", readonly=True)
-    valuation = fields.Selection(related="categ_id.property_valuation", readonly=True)
+    cost_method = fields.Selection(
```

Comment:
Missing conversion of following related fields from `related` to `compute` 

https://github.com/odoo/odoo/blob/60c6a52b8b9abfbad4fb47b4aaa6f22154877c80/addons/stock_account/models/stock_quant.py#L20

https://github.com/odoo/odoo/blob/60c6a52b8b9abfbad4fb47b4aaa6f22154877c80/addons/stock_account/wizard/stock_valuation_layer_revaluation.py#L39

https://github.com/odoo/odoo/blob/60c6a52b8b9abfbad4fb47b4aaa6f22154877c80/addons/stock_account/wizard/stock_valuation_layer_revaluation.py#L121